src/main/java/org/eclipsefoundation/adopters/service/impl/PagintationProjectsService.java

```diff
@@ -0,0 +1,107 @@
+/*******************************************************************************
+ * Copyright (C) 2020 Eclipse Foundation
+ * 
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ ******************************************************************************/
+package org.eclipsefoundation.adopters.service.impl;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+import org.eclipsefoundation.adopters.api.ProjectsAPI;
+import org.eclipsefoundation.adopters.model.Project;
+import org.eclipsefoundation.adopters.service.ProjectService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+
+import io.quarkus.runtime.Startup;
+
+/**
+ * Projects service implementation that handles pagination of data manually, as
+ * well as makes use of a loading cache to have data be always available with as
+ * little latency to the user as possible.
+ * 
+ * @author Martin Lowe
+ */
+@Startup
+@ApplicationScoped
+public class PagintationProjectsService implements ProjectService {
+	private static final Logger LOGGER = LoggerFactory.getLogger(PagintationProjectsService.class);
+
+	@Inject
+	@RestClient
+	ProjectsAPI projects;
+	// this class has a separate cache as this data is long to load and should be
+	// always available.
+	LoadingCache<String, List<Project>> internalCache;
```

Comment:
It's still a preview, but once GA, it may interesting to evaluate https://quarkus.io/guides/cache to replace the usage of guava cache

Reply:
I'm actually looking at adding that to the core lib that we're making! There are a few quirks but it seems to work alright. I'd need more load testing to ensure it can hold up to high amounts of reads, but it looks promising already